packages/calypso-e2e/src/lib/pages/reader-page.ts

```diff
@@ -0,0 +1,79 @@
+import { Page } from 'playwright';
+
+const selectors = {
+	readerVisitLink: '.reader-visit-link',
+	shareButtonLocator: '.reader-share__button',
+	firstComboCardPostLocator: '.reader-combined-card__post-title-link',
+	siteContentLocator: '.site-selector__sites .site__content',
+	commentButtonLocator: '.comment-button',
+	commentTextAreaLocator: '.comments__form textarea',
+	commentSubmitLocator: '.comments__form button',
+	commentContentLocator: '.comments__comment-content',
+};
+
+/**
+ * Page representing the Reader Page.
+ */
+export class ReaderPage {
+	private page: Page;
+
+	/**
+	 * Constructs an instance of the component.
+	 *
+	 * @param {Page} page The underlying page.
+	 */
+	constructor( page: Page ) {
+		this.page = page;
+	}
+
+	/**
+	 * Get the URL of the latest post in Reader
+	 *
+	 * @returns {Promise<string>} String of URL for latest post.
+	 */
+	async siteOfLatestPost(): Promise< string > {
+		const href = await this.page.getAttribute( selectors.readerVisitLink, 'href' );
+		return new URL( href ? href : '' ).host;
+	}
+
+	/**
+	 * Share latest post in Reader
+	 *
+	 * @returns {Promise<void>} No return value.
+	 */
+	async shareLatestPost(): Promise< void > {
+		const hasSharablePost = await this.page.$( selectors.shareButtonLocator );
+		if ( ! hasSharablePost ) {
+			// no shareable posts on this screen. try moving into a combined card
+			await this.page.click( selectors.firstComboCardPostLocator );
+		}
+
+		await this.page.click( selectors.shareButtonLocator );
+		await this.page.waitForSelector( selectors.siteContentLocator );
+		await this.page.click( selectors.siteContentLocator );
+	}
+
+	/**
+	 * Sets and submits comment on latest post
+	 *
+	 * @param {string} comment Text of the comment.
+	 * @returns {Promise<void>} No return value.
+	 */
+	async commentOnLatestPost( comment: string ): Promise< void > {
+		await this.page.click( selectors.commentButtonLocator );
+		await this.page.fill( selectors.commentTextAreaLocator, comment );
+		await this.page.click( selectors.commentSubmitLocator );
+	}
+
+	/**
+	 * Wait for the comment with the specified text to display.
+	 *
+	 * @param {string} comment Text of the comment.
+	 * @returns {Promise<void>} No return value.
+	 */
+	async waitForCommentToAppear( comment: string ): Promise< void > {
+		await this.page.waitForSelector(
+			selectors.commentContentLocator + ' :text("' + comment + '")'
```

Comment:
Completely optional!

One of the things Edwin and I have tried as a pattern in some tests is using the idea of a "dynamic selector" where the selector value in the constant values at the top of the module is actually a tiny little function that injects dynamic content! I thought that might potentially read nicely here :)